File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def solution(data, ext, val_ext, sort_by):
    # Define a dictionary to map the column names to their respective indices
    column_indices = {"code": 0, "date": 1, "maximum": 2, "remain": 3}
    
    # Filter the data based on the condition: ext value < val_ext
    filtered_data = [item for item in data if item[column_indices[ext]] < val_ext]
    
    # Sort the filtered data based on the sort_by criteria
    sorted_data = sorted(filtered_data, key=lambda x: x[column_indices[sort_by]])
    
    return sorted_data

File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.java
import java.util.*;
class Solution {
    public int[][] solution(int[][] data, String ext, int val_ext, String sort_by) {
        ArrayList<int[]> answer = new ArrayList<>();
        HashMap<String,Integer> colIdx = new HashMap<>();
        
        colIdx.put("code",0);
        colIdx.put("date",1);
        colIdx.put("maximum",2);
        colIdx.put("remain",3);
        for(int i=0;i<data.length;i++){
            if(data[i][colIdx.get(ext)]<val_ext){
                answer.add(data[i]);
            }
        }
        answer.sort(Comparator.comparingInt(a->a[colIdx.get(sort_by)]));
        
        return answer.toArray(new int[answer.size()][]);
    }
}